File: Front End/src/app/sector/sector.ts
import { INursery } from "../nursary/nursery";
import { ISapling } from "../sapling/sapling";

export interface ISector {
    id: string;
    number: number;
    capacity: number;
    currentQuantity: number;
    isActive: boolean;
    nurseryId: string;
    nursery?: INursery | null;
    saplings: ISapling[];
}

File: Back End/Application.Domain/Models/Sector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.Domain.Models
{
    // قطاع 
    public class Sector
    {
        public Guid Id { get; set; }
        public int Number { get; set; }
        public int Capacity { get; set; } // السعة 
        public int CurrentQuantity { get; set; }
        public bool IsActive { get; set; } = true;

        public Guid NurseryId { get; set; }
        public Nursery? Nursery { get; set; }
        public List<Sapling> Saplings { get; set; } = new List<Sapling>();


        public Sector()
        {
            Id = Guid.NewGuid();
        }
    }
}
